win/WIN_ButtonUtils.cpp

```diff
@@ -0,0 +1,50 @@
+#include "WIN_ButtonUtils.h"
+#include <SDL.h>
+#include <SDL_image.h>
+#include <cstdio>
+#include <cassert>
+
+using namespace win;
+
+SDL_Texture* ButtonUtils::loadSprite(SDL_Renderer* renderer, const char* path)
+{
+	//The final texture
+	SDL_Texture* newTexture = nullptr;
+
+	//Load image at specified path
+	auto* loadedSurface = IMG_Load(path);
+	if (loadedSurface == nullptr) {
+		printf("Unable to load image %s! SDL_image Error: %s\n", path, IMG_GetError());
+	}
+	else {
+		//Create texture from surface pixels
+		newTexture = SDL_CreateTextureFromSurface(renderer, loadedSurface);
+		if (newTexture == nullptr) {
+			printf("Unable to create texture from %s! SDL Error: %s\n", path, SDL_GetError());
+		}
+		//Get rid of old loaded surface
+		SDL_FreeSurface(loadedSurface);
+	}
+	return newTexture;
+}
+
+std::vector<SDL_Rect> ButtonUtils::handleSpriteSheet(SDL_Texture* texture)
+{
+	//Load sprite sheet texture
+	assert(texture);
+	std::vector<SDL_Rect> spriteClips;
+	
+	//Set top left sprite
+	spriteClips.push_back({ 0, 0, 100, 100 });
```

Comment:
So how do I know the dimensions are 100 x 100? Perhaps this should be passed in as a parameter, or rows x columns and the actual dimensions inferred by width / colums, height / rows and if they don't give exact multiples => error.
**MUST**

Reply:
done - uses image height and width and sets sprite dimensions using width/2 and height/2 (sprites must be in a 2x2 grid with the same dimensions as eachother).